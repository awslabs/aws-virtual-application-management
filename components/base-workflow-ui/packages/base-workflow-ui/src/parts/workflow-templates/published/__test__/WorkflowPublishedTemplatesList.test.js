/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import WorkflowPublishedTemplatesList from '../WorkflowPublishedTemplatesList';

jest.mock('../../WorkflowTemplateCard');
jest.mock('../../../workflow-common/ProgressPlaceholder');
describe('WorkflowPublishedTemplatesList', () => {
  const workflowTemplatesStore = { load: jest.fn(), startHeartbeat: jest.fn(), error: {}, list: [] };

  it('renders correctly', () =>
    expect(
      renderer
        .create(<WorkflowPublishedTemplatesList.WrappedComponent workflowTemplatesStore={workflowTemplatesStore} />)
        .toJSON(),
    ).toMatchSnapshot());
});
