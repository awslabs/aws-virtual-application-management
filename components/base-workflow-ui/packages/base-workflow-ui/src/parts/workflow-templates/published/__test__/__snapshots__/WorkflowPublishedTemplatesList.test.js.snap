// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowPublishedTemplatesList renders correctly 1`] = `
<div
  className="mb4"
>
  <div
    className="mb3 flex"
  >
    <h3
      className="ui header color-grey mt1 mb0 flex-auto"
    >
      <i
        aria-hidden="true"
        className="list alternate outline icon align-top"
        onClick={[Function]}
      />
      <div
        className="content left-align"
      >
        Published Workflow Templates
      </div>
    </h3>
  </div>
  <div
    className="p0"
  >
    <div
      className="ui negative message clearfix"
    >
      <div
        className="header"
      >
        A problem was encountered
      </div>
      <p>
        Hmm... something went wrong
      </p>
    </div>
  </div>
</div>
`;
